tests/jerry/array-prototype-splice.js

```diff
@@ -175,3 +175,174 @@ try {
   assert (e.message === "foo");
   assert (e instanceof ReferenceError);
 }
+
+try {
+  Array.prototype.splice.call(undefined);
+  assert(false);
+} catch (e) {
+  assert(e instanceof TypeError);
+}
+
+try {
+  var o = {};
+  Object.defineProperty(o, 'toString', { 'get' : function() { throw new ReferenceError("foo"); } });
+  var a = { length : o };
+  Array.prototype.splice.call(a, function() { });
+  assert(false);
+} catch (e) {
+  assert(e instanceof ReferenceError);
+}
+
+try {
+  var o = {};
+  Object.defineProperty(o, 'toString', { 'get' : function() { throw new ReferenceError("foo"); } });
+  [1, 2].splice(o);
+  assert(false);
+} catch (e) {
+  assert(e instanceof ReferenceError);
+}
+
+try {
+  var o = {};
+  Object.defineProperty(o, 'toString', { 'get' : function() { throw new ReferenceError("foo"); } });
+  [1, 2].splice(1, o);
+  assert(false);
+} catch (e) {
+  assert(e instanceof ReferenceError);
+}
+
+try {
+  var a = [1, 5, ,2];
+  Object.defineProperty(a, '2', { 'get' : function() { throw new ReferenceError("foo"); } });
+  a.splice(1, 7, 7, 7, 7, 7);
+  assert(false);
+} catch (e) {
+  assert(e instanceof ReferenceError);
+}
+
+var a = {length : 13};
+Array.prototype.splice.call(a, function () {delete a});
+assert(a.length === 0);
+
+try {
+  var a = [1, 5, 6, 7, 8, 5];
+  Object.defineProperty(a, '1', { 'get' : function() { throw new ReferenceError("foo"); } });
+  Object.defineProperty(a, '0', { 'get' : function() { throw new ReferenceError("foo"); } });
+  Object.defineProperty(a, '2', { 'get' : function() { throw new ReferenceError("foo"); } });
```

Comment:
Use different error messages, and assert in the catch block which error has been thrown.